src/libraries/System.Text.Json/Common/JsonSourceGenerationOptionsAttribute.cs

```diff
@@ -15,11 +15,58 @@ namespace System.Text.Json.Serialization
 #endif
     sealed class JsonSourceGenerationOptionsAttribute : JsonAttribute
     {
+        /// <summary>
+        /// Constructs a new <see cref="JsonSourceGenerationOptionsAttribute"/> instance.
+        /// </summary>
+        public JsonSourceGenerationOptionsAttribute() { }
+
+        /// <summary>
+        /// Constructs a new <see cref="JsonSourceGenerationOptionsAttribute"/> instance with a predefined set of options determined by the specified <see cref="JsonSerializerDefaults"/>.
+        /// </summary>
+        /// <param name="defaults">The <see cref="JsonSerializerDefaults"/> to reason about.</param>
+        /// <exception cref="ArgumentOutOfRangeException">Invalid <paramref name="defaults"/> parameter.</exception>
+        public JsonSourceGenerationOptionsAttribute(JsonSerializerDefaults defaults)
```

Comment:
In what situation is this public constructor useful?  You can't have a JsonSerializerDefaults in metadata, so someone would be manually constructing a new attribute?

Reply:
But who ends up using this constructor?  You can't write:
```C#
[JsonSourceGenerationOptions(JsonSerializerDefaults.Web)]
```
That won't compile.

Reply:
Oh, nevermind, JsonSerializerDefaults is an enum, I thought it was a struct.